sway/input/text_input.c

```diff
@@ -404,8 +407,29 @@ static void input_popup_set_focus(struct sway_input_popup *popup,
 		// surface. Layer surfaces get destroyed as part of the output being
 		// destroyed, thus also trickling down to popups.
 		popup->fixed_output = layer->layer_surface->output;
+	} else if (lock_surface) {
+		wl_signal_add(&lock_surface->surface->events.unmap,
+			&popup->focused_surface_unmap);
+
+		struct sway_layer_surface *lock = lock_surface->data;
+		if (lock == NULL) {
+			return;
+		}
+
+		relative_parent = lock->scene->tree;
+		popup->desc.view = NULL;
+
+		// we don't need to add an event here to NULL out this field because
+		// this field will only be initialized if the popup is part of a layer
+		// surface. Layer surfaces get destroyed as part of the output being
+		// destroyed, thus also trickling down to popups.
+		popup->fixed_output = lock->layer_surface->output;
 	} else {
 		struct sway_view *view = view_from_wlr_surface(surface);
+		// In the future there may be other lock surface been added, so we also need to check here.
+		if (view == NULL) {
+			return;
```

Comment:
Can we log a debug message here?

Reply:
sure

Reply:
It seems like there are typos in the log message (double space, and "foget").

Suggested log message: "Unsupported IME focus surface"

(It can also happen with any example I gave earlier, e.g. nested sub-surface, so it's not purely about shells.)